### DISCUSSION TITLE
Minimal 64x4 SD card

### DISCUSSION THREAD

**Discussion by hans61** [slu4coder/Minimal-64x4-Home-Computer#17] [Ideas]
After many attempts I now have working hardware for SD card support.
writing and reading sectors (CMD17, CMD24) works.
First results here, the next step should be a FAT32 library.

https://github.com/hans61/Minimal-64x4/tree/main/hardware/SD-Card

**Comment by TwoPigsToTheWind** [comment]
Wonderful, great work

**Comment by slu4coder** [comment]
Looks fantastic. I will definitely dive into this :-)

**Comment by hans61** [comment]
My current status:

https://github.com/hans61/Minimal-64x4/tree/main/hardware/expansion-compact

**Reply by TwoPigsToTheWind** [reply]
All of that on a 100mm by 100mm! That is surprisingly few ICs for all the added functions. Looks great.

**Reply by hellboy73** [reply]
I didn't even finish building my first Minimal64x4 but already excited to have such expansion board :) 
Sound, joystick, SD card support, those are essential features to enable Minimal to wider user base. If we add power-on reset circuit here, like someone suggested and we have all-you-ever-need expansion card. :D

**Reply by hans61** [reply]
There is a new version. I will publish it in the next few days.
I tried some simple power on reset circuits. But they didn't work. I could use a small attiny or pic. But I don't think that would be suitable. If anyone has an idea for a simple circuit I would be happy. I only have a little space left on the board.

**Reply by hellboy73** [reply]
See [here](https://github.com/slu4coder/Minimal-64x4-Home-Computer/discussions/30) kipshere used ADM706AN for power on reset.

**Reply by hans61** [reply]
There will be an extension with power on reset with a ne555. I have published my current status here:
https://github.com/hans61/Minimal-64x4-Expansion

**Reply by hellboy73** [reply]
great news :)

**Reply by hellboy73** [reply]
Hi Hans61
Thanks for your expansion board. I have build rev.7 and it works well when it comes to VSync and joystick reading and initial reset, all fine. I'm not sure if sound works as there is no software to test it. Can you share a simple piece of code for sound testing? On your GitHub maybe? it would be useful also for reference as I'm always learning on examples :)

**Reply by old61** [reply]
I have added more software and information on my Github.

[Minimal-64x4-Expansion](https://github.com/hans61/Minimal-64x4-Expansion)

**Reply by hellboy73** [reply]
thanks! it works!

**Reply by hellboy73** [reply]
I just wrote new initialization routine for your Expansion. 
It can be used with or without the card as it is testing VSync. If detected, initializes sound with BIOS-like beep :) 
if Vsync is not detected it doesn't do anything else.
[https://github.com/hellboy73/M64x4-initexp](https://github.com/hellboy73/M64x4-initexp)
Sharing it here as on expansion board repository couldn't find a comment section.
